src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -138,6 +144,25 @@ function useChatUtilsHook(): UseChatUtilsHook {
 
   const helperSaveChannelDetail = async (channel: ChannelList, response: any) => {
     if (!localDb) return;
+    response?.messages?.map((message) => {
+      const chatMessage = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+      return queue.addPriorityJob({
+        label: `saveChatMessage-${message?.id}`,
+        priority: QueueJobPriority.MEDIUM,
+        task: async () => {
+          await chatMessage.save(localDb);
+        }
+      });
+    });
+
+    queue.addPriorityJob({
+      label: 'refresh channel list',
+      priority: QueueJobPriority.MEDIUM,
+      task: async () => {
+        refresh('channelList');
+      }
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of a queue for saving chat messages and refreshing the channel list is a good approach for managing asynchronous tasks. However, the error handling could be improved. If an error occurs during the execution of a task, it should be handled appropriately.
```diff
    response?.messages?.map((message) => {
      const chatMessage = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
      return queue.addPriorityJob({
        label: `saveChatMessage-${message?.id}`,
        priority: QueueJobPriority.MEDIUM,
        task: async () => {
          try {
            await chatMessage.save(localDb);
+            } catch (e) {
+              console.error(`Failed to save chat message ${message?.id}`, e);
+            }
        }
      });
    });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->